### PR TITLE
Include transaction service && spec http approach

### PR DESCRIPTION
**Pull request by Franx0**
[DP-433](https://devengers.atlassian.net/browse/DP-433)

This PR includes:

1. transactions history service.
2. approach to run specs using http response files for transactions service
3. transactions -> http to run requests to Modulr locally (doc/modulr_api/**)

### CODE REVIEW

File: lib/modulr/resources/transactions/transaction.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        map :sourceId, :source_id
+        map :sourceExternalReference, :external_reference
+        map :additionalInfo, :additional_info
+        map :balance, :balance
+        map :available_balance, :available_balance
+
+        def initialize(response, attrs)
+          super
+          @balance = attrs[:account][:balance]
+          @available_balance = attrs[:account][:availableBalance]
```
Comment (Franx0) [inline lib/modulr/resources/transactions/transaction.rb:24]: Not pretty sure if we want to get current modulr account balance on each transaction 🤔 but I decided to receover it from response and included inside transaction object

File: spec/fixtures/transactions/history/responses/success.http [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Vary: Origin
+X-Content-type-options: nosniff
+X-Frame-Options: SAMEORIGIN
+X-Ratelimit-Limit: 200000
+X-Ratelimit-Remaining: 199993
+X-Ratelimit-Reset: 1661868222
+X-Xss-Protection: 1; mode=block
+
+{
+  "content": [
```
Comment (Franx0) [inline spec/fixtures/transactions/history/responses/success.http:19]: extracted from [Modulr API](https://modulr.readme.io/reference/gettransactionsbyaccount)

**Review by donmik — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -26,22 +26,51 @@ gem install modulr-api
 
 ## Usage
 
-TODO: Write usage instructions here
-
 ## Development
 
-After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+After checking out the repo, run `bin/setup` to install dependencies. Then, run `bundle exec rake` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
```
Comment (IvanGuardado) [inline README.md:31]: `bundle exec rake to run the tests.` is that right?

File: README.md [hunk 4]
```diff
@@ -26,22 +26,51 @@ gem install modulr-api
...
+
+To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, update the version number in `version.rb`, and then tag and push the new version:
```
Comment (IvanGuardado) [inline README.md:33]: Could we add this inside a `Release` section?

File: README.md [hunk 5]
```diff
@@ -26,22 +26,51 @@ gem install modulr-api
...
+git push origin vx.x.x
+```
+
+The tagging will trigger the GitHub action defined in `release.yml`, pushing the gem to [rubygems.org](https://rubygems.org).
+
+## Tooling
+
+We have a tooling system based on HTTP request files which allow developers to run pre-format Modulr API requests easily. These HTTP request files are stored inside `doc/modulr_requests` directory. Find the service you want to run and run the specific HTTP file to execute the request locally.
+
+We run these requests locally from our favorite IDLEs, in our case, we use ([IntelliJ](https://www.jetbrains.com/es-es/idea/) or [VS Code](https://code.visualstudio.com/)). [IntelliJ](https://www.jetbrains.com/es-es/idea/) has a native feature that allows you to run these requests easily but if you are using [VS Code](https://code.visualstudio.com/) you will need to install the following extension [httpYac](https://marketplace.visualstudio.com/items?itemName=anweber.vscode-httpyac).
```
Comment (IvanGuardado) [inline README.md:46]: Typo in `IDLEs`.

Comment (IvanGuardado) [inline README.md:46]: I am missing a brief explanation of how to configure the HTTP files environment. (the `http-client.private.env.json` file)

File: lib/modulr/resources/transactions/transaction.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        map :credit, :credit
+        map :type, :type
+        map :sourceId, :source_id
+        map :sourceExternalReference, :external_reference
+        map :additionalInfo, :additional_info
+        map :balance, :balance
+        map :available_balance, :available_balance
+
+        def initialize(response, attrs)
+          super
```
Comment (IvanGuardado) [inline lib/modulr/resources/transactions/transaction.rb:22]: Don't we have to pass the response to the `super`?

**Pull request merged**